includes/class-freemius.php

```diff
@@ -20992,22 +20992,38 @@ private function _sync_plugin_license(
                              */
                             $api = $this->get_api_site_scope();
 
-                            if ( ! self::$_global_admin_notices->has_sticky( 'api_blocked' ) ) {
-                                self::$_global_admin_notices->add(
-                                    sprintf(
-                                        $this->get_text_inline( 'Your server is blocking the access to Freemius\' API, which is crucial for %1$s synchronization. Please contact your host to whitelist %2$s', 'server-blocking-access' ),
-                                        $this->get_plugin_name(),
-                                        '<b>' . implode( ', ', $this->apply_filters( 'api_domains', array(
-                                            'api.freemius.com',
-                                            'wp.freemius.com'
-                                        ) ) ) . '</b>'
-                                    ) . '<br> ' . $this->get_text_inline( 'Error received from the server:', 'server-error-message' ) . var_export( $result->error, true ),
-                                    $this->get_text_x_inline( 'Oops', 'exclamation', 'oops' ) . '...',
-                                    'error',
-                                    $background,
-                                    'api_blocked'
-                                );
-                            }
+	                        $add_notice = false;
+	                        if ( $background ) {
+		                        $counter = (int) get_transient( '_fs_api_connection_retry_counter' );
+
+		                        if ( $counter >= 3 ) {
+			                        $add_notice = true;
+			                        $counter    = 0;

```

Comment:
@msladu 
1. Resetting the counter after 3 failures is a good idea, but it's not obvious and requires an explanation that we only want to show the notice after 3 consecutive failures, and then if the notice was already shown, start the counter from scratch.
2. In addition to resetting it here, we also need to reset the counter (actually delete the transient) when the API request was valid. Otherwise, if the API fails twice, then succeeds, and then fails one more time, the message will be shown, even though there was only one consecutive connectivity failure and not 3.

Reply:
@vovafeldman  

1. Yes, I will add the explanation. 
2. Got it. Will implement that.

Reply:
@msladu you didn't add clarification about why the counter is reset after 3 attempts (the first comment) in this thread.

Reply:
@vovafeldman Ok. I am adding that now.